In [8]:
# HIDDEN
Base.displaysize() = (5, 110)
using DataFrames
using CSV

In [2]:
# HIDDEN
calls = CSV.read("data/calls_julia.csv")
stops = CSV.read("data/stops_julia.csv");

## Scope

The scope of the dataset refers to the coverage of the dataset in relation to what we are interested in analyzing. We seek to answer the following question about our data scope:

**Does the data cover the topic of interest?**

For example, the Calls and Stops datasets contain call and stop incidents made in Berkeley. If we are interested in crime incidents in the state of California, however, these datasets will be too limited in scope.

In general, larger scope is more useful than smaller scope since we can filter larger scope down to a smaller scope but often can't go from smaller scope to larger scope. For example, if we had a dataset of police stops in the United States we could subset the dataset to investigate Berkeley.

Keep in mind that scope is a broad term not always used to describe geographic location. For example, it can also refer to time coverage — the Calls dataset only contains data for a 180 day period.

We will often address the scope of the dataset during the investigation of the data generation process and confirm the dataset's scope during EDA. Let's confirm the geographic and time scope of the Calls dataset.

In [3]:
calls

5508×8 DataFrame

We see that EVENTDTTM is of type `String`. For being able to perform operations on that column it would be better to extract it to another column as the type `Date`:

In [17]:
using Dates
# Split the EVENTDTTM string and select only the date
calls.DATE = first.(split.(calls.EVENTDTTM))
# Transform it into Date
calls.DATE = Date.(calls.DATE, dateformat"m/d/y")
calls

5508×10 DataFrame

In [33]:
# Shows earliest and latest dates in calls
sorted_dates = sort(calls, :DATE)
println(first(sorted_dates, 5).DATE)
println(last(sorted_dates, 5).DATE)

Date[2017-03-02, 2017-03-02, 2017-03-02, 2017-03-02, 2017-03-02]
Date[2017-08-27, 2017-08-27, 2017-08-28, 2017-08-28, 2017-08-28]


In [36]:
sorted_dates[end, :DATE] - sorted_dates[1, :DATE]

179 days

The table contains data for a time period of 179 days which is close enough to the 180 day time period in the data description that we can suppose there were no calls on either April 14st, 2017 or August 29, 2017.

To check the geographic scope, we can use a map:

-- NEED to Plot!! --

With a few exceptions, the Calls dataset covers the Berkeley area. We can see that most police calls happened in the Downtown Berkeley and south of UC Berkeley campus areas.

Let's now confirm the temporal and geographic scope for the Stops dataset:

In [37]:
stops

29208×7 DataFrame

In [42]:
sort(stops, Symbol("Call Date/Time"))

29208×7 DataFrame

As promised, the data collection begins on January 26th, 2015. It looks like the data were downloaded somewhere around the beginning of May 2017 since the dates stop on April 30th, 2017. Let's draw a map to see the geographic data:

--NEED TO PLOT--

We can confirm that the police stops in the dataset happened in Berkeley, and that most police calls happened in the Downtown Berkeley and West Berkeley areas.